lib/bundle_update_interactive/changelog_locator.rb

```diff
@@ -5,8 +5,8 @@
 
 GITHUB_PATTERN = %r{^(?:https?://)?github\.com/([^/]+/[^/]+)(?:\.git)?/?}.freeze
 URI_KEYS = %w[source_code_uri homepage_uri bug_tracker_uri wiki_uri].freeze
-FILE_PATTERN = /(?:changelog|changes|history|news|release)/.freeze
-EXT_PATTERN = /(?:md|txt|rdoc)/.freeze
+FILE_PATTERN = /changelog|changes|history|news|release/i.freeze
+EXT_PATTERN = /md|txt|rdoc/i.freeze
```

Comment:
🗒️ TIL that when a regex is embedded within a regex literal in Ruby, like `/#{pattern}/i`, Ruby automatically resets the matching options such that they don't apply to the embedded regex. In other words, even though the outer regex literal specifies `/i`, this option will _not_ apply to the embedded regex unless that regex also declares `/i`.

In the context of this PR, I thought `FILE_PATTERN` and `EXT_PATTERN` were being interpreted as case-insensitive via the rexep in which they were embedded, but testing showed this was not the case.

Fix by explicitly adding `/i` to these patterns.

I also removed the `?:` wrapper because Ruby adds it automatically when a regex is embedded in another regex literal.